sc2/unit.py

```diff
@@ -295,6 +295,121 @@ def movement_speed(self) -> float:
+                speed *= 1.6
+            elif unit_type == UnitTypeId.BANELING and UpgradeId.CENTRIFICALHOOKS in upgrades:
+                speed += 0.63
+            '''
+            elif unit_type == UnitTypeId.LURKERMP and UpgradeId.DIGGINGCLAWS in upgrades:
+                pass  # todo: can't find how much it increases speed
+            '''
+            ''' todo: add this in 4.11:
+
+            elif unit_type == UnitTypeId.VOIDRAY and UpgradeId.VOIDRAYSPEEDUPGRADE in upgrades:
+                speed += 0.703
+
+            # "Rapid Reignition System" upgrade increases Medivac's base movement speed from 3.5 to 4.13
+            # doesn't increase speed after boost
+            elif unit_type == UnitTypeId.MEDIVAC and UpgradeId.MEDIVACRAPIDDEPLOYMENT in upgrades:
+                speed = 4.13
+
+            # In 4.11 Charge also increases Zealot's base speed to 4.725
+            elif unit_type == UnitTypeId.ZEALOT and UpgradeId.CHARGE in upgrades:
+                speed = 4.725
+            '''
+
+        # ---- Creep ----
+        if not self.is_flying:
+            on_creep: bool = self._bot_object.state.creep[self.position.rounded] == 1
+
+            if on_creep:
+                if unit_type in {
+                        UnitTypeId.ZERGLING,
+                        UnitTypeId.BANELING,
+                        UnitTypeId.ROACH,
+                        UnitTypeId.RAVAGER,
+                        UnitTypeId.HYDRALISK,
+                        UnitTypeId.LURKERMP,
+                        UnitTypeId.ULTRALISK,
+                        UnitTypeId.INFESTOR,
+                        UnitTypeId.INFESTORTERRAN,
+                        UnitTypeId.SWARMHOSTMP,
+                    }:
+                    speed *= 1.3
+                elif unit_type == UnitTypeId.QUEEN:
+                    # 167% bonus
+                    speed *= 2.67
+                elif unit_type == UnitTypeId.LOCUSTMP:
+                    speed *= 1.4
+                elif unit_type in {UnitTypeId.SPINECRAWLER, UnitTypeId.SPORECRAWLER}:
+                    # 150% bonus
+                    speed *= 2.5
+
+            # off creep
+            elif upgrades:
+                if unit_type == UnitTypeId.HYDRALISK and UpgradeId.EVOLVEMUSCULARAUGMENTS in upgrades:
+                    speed *= 1.25
+                '''
+                elif unit_type == UnitTypeId.ULTRALISK and UpgradeId.ANABOLICSYNTHESIS in upgrades:
+                    pass  # todo: can't find how much it increases speed
+                '''
+
+            # Ultralisk has passive ability "Frenzied" which mades it immune to speed altering buffs
+            if unit_type == UnitTypeId.ULTRALISK:
+                return speed
+
+        # ---- Buffs ----
+        buffs = self.buffs
+
+        if not {BuffId.STIMPACK, BuffId.STIMPACKMARAUDER}.isdisjoint(buffs):
+            speed *= 1.5
+        elif BuffId.MEDIVACSPEEDBOOST in buffs:
+            speed = 5.94
```

Comment:
Is this the value from the game or from the API?
As far as I know (but I could be wrong):
`API value = 1.4 * game (or liquipedia) value`

Reply:
Value is from game, not api.
Since medivac's base speed = 3.5 and it increases to 5.94, increase factor is about 1.7 (5.94 / 3.5 = 1.6971428571428573).
Then you might replace this code to `speed = self.movement_speed * 1.7` or even `speed = self.movement_speed * 1.6971428571428573`